Proposal.md

```diff
@@ -0,0 +1,604 @@
+# Add RangeSet and Related Collection Operations
```

Comment:
Agreed on the title, I've switched it around a bit. I pulled rotation and partition out to simplify this proposal, and intend to pitch them separately so that we can give them the discussion they deserve, instead of having both discussion during this review.